Movie profile on ID

Mattan41 opened the pull request:
### Summary
This pull request includes several updates and improvements to the movie application.

### Changes
1. **Fetch Movies in App.vue**:
   - Added functionality to fetch the list of movies in `App.vue`.
   - Passed the fetched list as props via the router to `LandingView`.

2. **Update Genre Handling in MovieProfileView**:
   - Improved the handling of genres in `MovieProfileView`.

3. **Update Backend Endpoint**:
   - Updated the backend endpoint when fetching a single movie by its ID.

4. **Refactor LandingView.vue**:
   - Refactored `LandingView.vue` to use `RouterLink` for navigation.
   - Updated routing to navigate to `MovieProfileView` with `id` and `title` parameters.

Mattan41 committed:
add fetch movies in app.vue, pass the list as props via router to LandingView.
update imgUrl with https://image.tmdb.org/t/p/w500 before movie.url

Mattan41 committed:
send movie as prop Json.stringified via routerlink and router

Mattan41 committed:
update genre-handling in MovieProfilView

Mattan41 committed:
update endpoint to get one movie on id

Mattan41 committed:
MovieProfileView.vue
fetch movie based on movie Id instead of passing object as a prop
add a v-if check to ensure that the movie object is not null before rendering its properties.

LandingView.vue, replace % with hyphens in movie-title for url

update props: route to pass id and title of movie.

DevelopedByDanny committed:
show all movies on landingpage

Mattan41 committed:
Refactor Top 10 movies in LandingView.vue to use RouterLink and handle undefined properties

Frlundsten commented on frontend/package.json:
```diff
@@ -15,7 +15,7 @@
     "tailwindcss": "^3.4.13",
     "vue": "^3.4.29",
-    "vue-router": "^4.3.3"
+    "vue-router": "^4.0.13"
```
I think this is an older version than the previous.

Mattan41 committed:
MovieProfileView.spec.js update genres in dummydata to match our database. Mock fetch movie before each tests

Mattan41 committed:
LandingView.spec.js update genres and imgurl in dummydata to match our database.

Mattan41 committed:
LandingView.spec.js add RouterLinkStub

Mattan41 committed:
update to latest vue-router

DevelopedByDanny committed:
show all movies on landingpage again

Mattan41 committed:
Merge remote-tracking branch 'origin/showMovieList' into movieProfile-id

# Conflicts:
#	frontend/src/router/index.js
#	frontend/src/views/LandingView.vue

Mattan41 committed:
update arrayOfHighestRated to use movies.value

Mattan41 committed:
update LandingView.spec.js to mock wrapper.vm.movies = movieArr.value; instead of props

Mattan41 committed:
Update LandingView.vue movie.genre to movie.genres

This commit fixes the display of genres in the list of movies on LandningView.vue

Frlundsten reviewed (approved):
💯 Good job!

Emmelie83 reviewed (approved):
I very much approve :-)
